internal/reports/csv.go

```diff
@@ -0,0 +1,437 @@
+package reports
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/kubev2v/migration-planner/internal/store/model"
+	"github.com/kubev2v/migration-planner/api/v1alpha1"
+)
+
+// GenerateCSVContent generates CSV content from source.Inventory data
+func GenerateCSVContent(source *model.Source, reportType string, includeWarnings bool) string {
```

Comment:
please return an error when generating the HTML fails. It should be a service error that handler can translate into HTTP error: 400 404 500